Library/Homebrew/cmd/doctor.rb

```diff
@@ -419,6 +419,16 @@ def check_access_usr_local
   end
 end
 
+def check_tmpdir_sticky_bit
+  world_writable = HOMEBREW_TEMP.stat.mode & 0777 == 0777
+  if world_writable && !HOMEBREW_TEMP.sticky? then <<-EOS.undent
+    #{HOMEBREW_TEMP} is not sticky. This is likely to cause formulae to
+    fail to compile. Please run "Repair Disk Permissions" in Disk Utility.
+  EOS
+  end
```

Comment:
Can we reword this a bit:

>#{HOMEBREW_TEMP} is world-writable but does not have the sticky bit set.
Please run "Repair Disk Permissions" in Disk Utility.

The reasoning being that this won't actually cause compilation failures, it will cause ruby to emit a error out when it is trying to create the temporary build directory.